daemon/api_sideload_n_try.go

```diff
@@ -522,10 +529,10 @@ func sideloadSnap(_ context.Context, st *state.State, upload *uploadedContainer,
 
 	model := deviceCtx.Model()
 
-	sideInfo, snapErr := readSideInfo(st, upload.tmpPath, upload.filename, flags, model)
+	info, snapErr := readInfoAndDeriveSideInfo(st, upload.tmpPath, upload.filename, flags, model)
 	if snapErr != nil {
```

Comment:
maybe this need an explicit comment that we are trying for a component here? we are also going to hash the filed twice, right? maybe we need a TODO about that